lib/ledger_sync/adaptors/quickbooks_online/account/operations/find.rb

```diff
@@ -0,0 +1,35 @@
+module LedgerSync
+  module Adaptors
+    module QuickBooksOnline
+      module Account
+        module Operations
+          class Find < Operation::Find
+            class Contract < LedgerSync::Adaptors::Contract
+              params do
+                required(:ledger_id).filled(:string)
+                required(:name).maybe(:string)
+                required(:account_type).maybe(:string)
+                required(:account_sub_type).maybe(:string)
+              end
+            end
+
+            private
+
+            def operate
+              return failure(nil) if resource.ledger_id.nil?
+
+              response = @adaptor.find(
+                resource: 'account',
+                id: resource.ledger_id
+              )
+
+              success(response: response)
+            rescue OAuth2::Error => e
```

Comment:
This would need to be refactored bit better then just moving it above to adaptor. Basically adaptor would need to return object that either does success or failure and `operate` would then return result based on that. We would still have condition here and we would still be returning `success` or `failure`. The only difference is that we wouldnt be catching `Oauth` error in operation. Which kinda makes sense. But on the other side, its not like adaptor operations are adaptor independent.